libs/wingsdk/test/target-tf-aws/bucket.test.ts

```diff
@@ -113,3 +125,119 @@ test("bucket prefix can not begining with 'xn--'", () => {
     /AWS S3 bucket names cannot begin with 'xn--'/
   );
 });
+
+test("bucket with onCreate method", () => {
+  // GIVEN
+  const app = new tfaws.App({ outdir: mkdtemp() });
+  const bucket = Bucket._newBucket(app, "my_bucket", { public: true });
+  const inflightTest = new InflightBucketEventHandler(app, "inflight");
+  bucket.onCreate(inflightTest);
+  const output = app.synth();
+
+  // THEN
+  expect(tfResourcesOf(output)).toEqual([
+    "aws_iam_role",
+    "aws_iam_role_policy",
+    "aws_iam_role_policy_attachment",
+    "aws_lambda_function", // inflight subscriber
+    "aws_lambda_permission", // permission of the topic to invoke lambda
+    "aws_s3_bucket", // main bucket
+    "aws_s3_bucket_notification",
+    "aws_s3_bucket_policy", // resource policy to grant read access to anyone
+    "aws_s3_bucket_server_side_encryption_configuration", // server side encryption
+    "aws_s3_object",
+    "aws_sns_topic", // topic to subscribe to bucket events
+    "aws_sns_topic_policy", //permission of the bucket to publish events
+    "aws_sns_topic_subscription", // subscription to events
+  ]);
```

Comment:
Maybe for this section we can just assert about the specific resources that are related to `onCreate`, for example:

```
expect(tfResourcesOfCount(output, "aws_sns_topic")).toEqual(1);
expect(tfResourcesOfCount(output, "aws_s3_bucket_notification")).toEqual(1);
// ...
```